extensions/plugins/system/redcore/redcore.php

```diff
@@ -164,6 +166,43 @@ public function onAfterInitialise()
 	}
 
 	/**
+	 * Method to clear the cache and any session state
+	 * related to API requests
+	 *
+	 * @param   string   $group      The cache group
+	 * @param   integer  $client_id  The ID of the client
+	 *
+	 * @throws Exception
+	 * @return void
+	 */
+	private function cleanCache($group = null, $client_id = 0)
+	{
+		$input = JFactory::getApplication()->input;
+		$params = $this->params;
+
+		if ($input->get('stateful', $params->get('stateful_webservice', 0)) == 1)
```

Comment:
also more complex URI param should be used, if one at all... I prefer using headers for options but URI is ok too I guess. It needs to be webservice_statefull or something like that because it will not collide with other URI parameters